datenguide_python/query_builder.py

```diff
@@ -351,8 +523,41 @@ def get_fields(self) -> List[str]:
         """
         return self.start_field.get_fields()
 
-    def results(self) -> Optional[ExecutionResults]:
-        return QueryExecutioner().run_query(self)
+    def results(self) -> Union[DataFrame, Optional[ExecutionResults]]:
+        """Runs the query and returns a Pandas DataFrame with the results.
+
+        Raises:
+            ValueError: If the Query did not return any results.
+            E.g. if the Query was ill-formed.
+
+        Returns:
+            Union[DataFrame, Optional[ExecutionResults]] --
+            A DataFrame with the queried data.
+            If the query fails raise ValueError.
+        """
+        result = QueryExecutioner().run_query(self)
+        if result:
+            return QueryOutputTransformer(result.query_results).transform()
+        else:
+            raise ValueError("No results could be returned for this Query.")
+
+    def meta_data(self) -> Optional[Dict[str, Any]]:
+        """Runs the query and returns a Dict with the meta data of the queries results.
+
+        Raises:
+            ValueError: If the Query did not return any results.
+            E.g. if the Query was ill-formed.
+
+        Returns:
+            Union[DataFrame, Optional[ExecutionResults]] --
+            A Dict with the queried meta data.
+            If the query fails raise ValueError.
+        """
+        result = QueryExecutioner().run_query(self)
+        if result:
+            return result.meta_data
+        else:
+            raise ValueError("No results could be returned for this Query.")
```

Comment:
Again unsure about the error type. My goto error when in Doubt is RuntimeError but more specific errors are of course better.